src/components/swap/SwapHeader.tsx

```diff
@@ -13,14 +22,120 @@ const StyledSwapHeader = styled.div`
   color: ${({ theme }) => theme.textSecondary};
 `
 
+const Dot = styled.div`
+  height: 8px;
+  width: 8px;
+  background-color: ${({ theme }) => theme.accentActive};
+  border-radius: 50%;
+`
+
+const TextHeader = styled.div<{ color: string; marginLeft?: string; isClickable?: boolean; disabled?: boolean }>`
+  color: ${({ color }) => color};
+  opacity: ${({ disabled }) => disabled && '60%'};
+  margin-left: ${({ marginLeft }) => marginLeft};
+  margin-right: 8px;
+  display: flex;
+  line-height: 20px;
+  flex-direction: row;
+  cursor: ${({ isClickable }) => isClickable && 'pointer'};
+  justify-content: center;
+  align-items: center;
+`
+
+const MOONPAY_REGION_AVAILABILITY_ARTICLE =
+  'https://support.uniswap.org/hc/en-us/articles/11306664890381-Why-isn-t-MoonPay-available-in-my-region-'
+
+// TODO(lynnshaoyu): add analytics and logging
 export default function SwapHeader({ allowedSlippage }: { allowedSlippage: Percent }) {
+  const theme = useTheme()
+  const { account } = useWeb3React()
+  const openFiatOnRampModal = useOpenModal(ApplicationModal.FIAT_ONRAMP)
+  const fiatOnRampButtonEnabled = useFiatOnRampButtonEnabled()
+  const [buyFiatClicked, setBuyFiatClicked] = useBuyFiatClicked()
+  const [continueBuyFiatFlowPostRegionCheck, setContinueBuyFiatFlowPostRegionCheck] = useState(false)
```

Comment:
all of this new code should go in a separate `Buy` button component (up til line 138) so that the SwapHeader file itself can be relatively straightforward, and the FOR logic can be isolated to its own thing

Reply:
sounds good done